e2e/testcases/helm_sync_test.go

```diff
@@ -178,6 +185,100 @@ func TestHelmDefaultNamespace(t *testing.T) {
 	}
 }
 
+// TestHelmLatestVersion verifies the Config Sync behavior for helm charts when helm.spec.version is not specified. The helm-sync
+// binary should pull down the latest available version in this case. It also tests that if a new helm chart gets pushed, the
+// chart version gets automatically updated by Config Sync.
+// This test will work only with following pre-requisites:
+// Google service account `e2e-test-ar-reader@${GCP_PROJECT}.iam.gserviceaccount.com` is created with `roles/artifactregistry.reader` for accessing images in Artifact Registry.
+// A JSON key file is generated for this service account and stored in Secret Manager
+func TestHelmLatestVersion(t *testing.T) {
+	nt := nomostest.New(t,
+		nomostesting.SyncSource,
+		ntopts.Unstructured,
+		ntopts.RequireGKE(t),
+	)
+
+	rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+	nt.T.Log("Fetch password from Secret Manager")
+	key, err := gitproviders.FetchCloudSecret("config-sync-ci-ar-key")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	user := "_json_key"
+
+	nt.T.Log("Create secret for authentication")
+	_, err = nt.Shell.Kubectl("create", "secret", "generic", "foo", fmt.Sprintf("--namespace=%s", v1.NSConfigManagementSystem), fmt.Sprintf("--from-literal=username=%s", user), fmt.Sprintf("--from-literal=password=%s", key))
+	if err != nil {
+		nt.T.Fatalf("failed to create secret, err: %v", err)
+	}
+	nt.T.Cleanup(func() {
+		nt.MustKubectl("delete", "secret", "foo", "-n", v1.NSConfigManagementSystem, "--ignore-not-found")
+	})
+
+	nt.T.Log("Update RootSync to sync from a private Artifact Registry")
+	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceType": "%s", "git": null, "helm": {"repo": "%s", "chart": "%s", "auth": "token", "version": "", "namespace": "", "deployNamespace": "", "secretRef": {"name" : "foo"}}}}`,
+		v1beta1.HelmSource, privateARHelmRegistry, privateSimpleHelmChart))
+	err = nt.WatchForAllSyncs(nomostest.WithRootSha1Func(helmChartVersion(privateSimpleHelmChart+":"+privateSimpleHelmChartLatestVersion)),
+		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: privateSimpleHelmChart}))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Validate("deploy-2.5.3", "default", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Push a new chart version to the private Artifact Registry")
+	token, err := exec.Command("gcloud", "auth", "print-access-token").CombinedOutput()
```

Comment:
Does `gcloud auth configure-docker us-docker.pkg.dev` work? From the [helm](https://cloud.google.com/artifact-registry/docs/helm/authentication#methods)/[gcloud](https://cloud.google.com/artifact-registry/docs/docker/authentication#gcloud-helper) docs it sounds like helm should work with the credential helper.

I think it would be preferred to avoid printing/storing an access token to minimize the risk of leaking a token. Using a credential helper is the most secure option AFAICT

Reply:
I tried `gcloud auth configure-docker us-docker.pkg.dev` and it didn't work. If you have other suggestions or know how to get that working I'm happy to try it out, but so far the PAT is the most secure option that I've been able to get to work (and FWIW they are short-lived).